Quad4.cs

```diff
@@ -0,0 +1,619 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using ISAAR.MSolve.PreProcessor.Embedding;
+using ISAAR.MSolve.PreProcessor.Interfaces;
+using ISAAR.MSolve.Matrices.Interfaces;
+using System.Runtime.InteropServices;
+using ISAAR.MSolve.Matrices;
+using ISAAR.MSolve.PreProcessor.Elements.SupportiveClasses;
+
+namespace ISAAR.MSolve.PreProcessor.Elements
+{
+	public class Quad4 : IStructuralFiniteElement, IEmbeddedHostElement
+	{
+		protected static double determinantTolerance = 0.00000001;
+
+		protected static int iInt = 2;
+		protected static int iInt2 = iInt * iInt;
+		protected static int iInt3 = iInt2 * iInt;
+        protected readonly static DOFType[] nodalDOFTypes = new DOFType[] { DOFType.X, DOFType.Y}; //Panos - 2 DoF per node
+        protected readonly static DOFType[][] dofTypes = new DOFType[][] { nodalDOFTypes, nodalDOFTypes, 
+            nodalDOFTypes,nodalDOFTypes}; //Panos - 4 nodes only
+		
+		//protected readonly IFiniteElementMaterial3D[] materialsAtGaussPoints; //Panos - What about 2D Material? Implemented
+		protected readonly IFiniteElementMaterial2D[] materialsAtGaussPoints; 
+
+		protected IFiniteElementDOFEnumerator dofEnumerator = new GenericDOFEnumerator();
+        protected Quad4()
+		{
+		}
+        public Quad4(IFiniteElementMaterial2D material)
+		{
+			materialsAtGaussPoints = new IFiniteElementMaterial2D[iInt2];
+			for (int i = 0; i < iInt2; i++)
+				materialsAtGaussPoints[i] = (IFiniteElementMaterial2D)material.Clone();
+		}
+        public Quad4(IFiniteElementMaterial2D material, IFiniteElementDOFEnumerator dofEnumerator)
+            : this(material)
+        {
+			this.dofEnumerator = dofEnumerator;
+		}
+        public IFiniteElementDOFEnumerator DOFEnumerator
+		{
+			get { return dofEnumerator; }
+			set { dofEnumerator = value; }
+		}
+        //Panos - I dont think I need these
```

Comment:
These properties should be put to an interface that will need to be implemented for elements that support dynamic analysis